aries_vcx/src/error.rs

```diff
@@ -358,6 +359,49 @@ impl From<AgencyClientErrorKind> for VcxErrorKind {
     }
 }
 
+impl From<messages::error::VcxError> for VcxError {
+    fn from(msg_err: messages::error::VcxError) -> VcxError {
+        let vcx_error_kind: VcxErrorKind = msg_err.kind().into();
+        VcxError::from_msg(vcx_error_kind, msg_err.to_string())
+    }
+}
+
+impl From<MessagesErrorKind> for VcxErrorKind {
```

Comment:
Maybe for separate PR, but this should be quite reworked - numeric error codes, many of the error kinds, C string definitions, all these things don't really make sense for this crate